src/commands/cmd_query.c

```diff
@@ -72,14 +72,18 @@ void Graph_Query(void *args) {
 	AST *ast = NULL;
 	bool cached = false;
 	ExecutionPlan *plan = NULL;
-	ExecutionType exec_type = ExecutionInformation_FromQuery(command_ctx->query, &plan, &ast, &cached);
+	ExecutionCtx exec_ctx = ExecutionCtx_FromQuery(command_ctx->query);
 	// See if there were any query compile time errors
 	if(QueryCtx_EncounteredError()) {
 		QueryCtx_EmitException();
 		goto cleanup;
 	}
+	ExecutionType exec_type = exec_ctx.exec_type;
 	if(exec_type == EXECUTION_TYPE_INVALID) goto cleanup;
```

Comment:
Do we need to check exec_type, isn't the test above line 77 is enough?

Reply:
safety